### PR TITLE
Implement BleHciSocket Modlue template

### PR DESCRIPTION
**Pull request by hs0225**
- Delete `BLE` native implementation.
- Implement `BLEHciSocket` native and js template.

This PR is initial implementation for BLE and BLEHciSocket module.

IoT.js-DCO-1.0-Signed-off-by: Hosung Kim hs852.kim@samsung.com

### CODE REVIEW

File: src/module/iotjs_module_blehcisocket.c [hunk 1]
```diff
@@ -0,0 +1,210 @@
+/* Copyright 2015-present Samsung Electronics Co., Ltd. and other contributors
```
Comment (nova0821) [inline src/module/iotjs_module_blehcisocket.c]: Copyright year should be `2017-present`.

File: src/module/iotjs_module_blehcisocket.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/* Copyright 2016-present Samsung Electronics Co., Ltd. and other contributors
```
Comment (nova0821) [inline src/module/iotjs_module_blehcisocket.h]: ditto

File: src/platform/arm-nuttx/iotjs_module_blehcisocket-arm-nuttx.c [hunk 3]
```diff
@@ -0,0 +1,93 @@
+/* Copyright 2016-present Samsung Electronics Co., Ltd. and other contributors
```
Comment (nova0821) [inline src/platform/arm-nuttx/iotjs_module_blehcisocket-arm-nuttx.c]: ditto

File: src/platform/iotjs_module_blehcisocket-linux-general.inl.h [hunk 4]
```diff
@@ -0,0 +1,100 @@
+/* Copyright 2016-present Samsung Electronics Co., Ltd. and other contributors
```
Comment (nova0821) [inline src/platform/iotjs_module_blehcisocket-linux-general.inl.h]: ditto

**Comment by hs0225** [comment]
@nova0821 I fixed it as your comments.

**Comment by glistening** [comment]
Personally I don't like putting much time to keep the copyright year number. Although it may be beneficial to someone, it is very tedious and need time to update correctly. So I changed the copyright  the ending year from explicit number to `present`. I think `2015-present` is is enough. `2015` is the starting year of _iotjs_ open source. If someone want to find out the starting date of the source they read, he can use `git log` command.

**Comment by nova0821** [comment]
@glistening I agreed it is not necessary to use much time to keep the copyright year, but it was our rule and I think it will be better to make as an issue about it if you want to change it.

**Comment by nova0821** [comment]
LGTM

**Pull request merged**